tritonsrc/attn_torch_function.py

```diff
@@ -474,6 +466,8 @@ def backward(ctx, do, _, fwd_tuning_result):
         else:
             BLOCK_M = 128
             BLOCK_N = 64
+        if q.dtype == torch.float32:
+            BLOCK_M //= 2
```

Comment:
I take it this was from memory limitations?

Reply:
Yes, but unfortunately this doesn't fix all problems.
fp32 gets broken in other places.